### PR TITLE
UAR-1561 Correct display of trustees for ceased trust

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-1561

### Change description

Updated the templates with more detailed conditions for displaying trustee "still involved" fields.

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/controllers/update/check.your.answers.controller.spec.ts [hunk 1]
```diff
@@ -968,7 +968,89 @@ describe("CHECK YOUR ANSWERS controller", () => {
...
+      };
+
+      mockGetApplicationData.mockReturnValue(appData);
+      mockIsActiveFeature.mockReturnValueOnce(true).mockReturnValueOnce(true).mockReturnValueOnce(true).mockReturnValueOnce(true);
+      const resp = await request(app).get(UPDATE_CHECK_YOUR_ANSWERS_URL);
+
       expect(resp.text).toContain(TRUSTEE_INDIVIDUAL_INVOLVED);
+      expect(resp.text).toContain(TRUSTEE_INDIVIDUAL_CEASED_DATE);
+      expect(resp.text).toContain("10");
+      expect(resp.text).toContain("May");
```
Comment (markpit) [inline test/controllers/update/check.your.answers.controller.spec.ts:1051]: the title of this test might be wrong as it says that the trustee still involved fields should be excluded but the test is expecting them to be present.

File: views/includes/check-your-answers/trusts.html [hunk 2]
```diff
@@ -18,8 +18,8 @@ <h3 class="govuk-heading-m">{{ trust_heading }}</h3>
   {% if isRegistration  %}
     {% set changeLinkUrl = OE_CONFIGS.TRUST_DETAILS_URL %}
   {% else %}
-    {% set stillInvolved = trust.trust_still_involved_in_overseas_entity %}
-    {% set isNotStillInvolved = stillInvolved === "No" %}    
+    {% set trustStillInvolved = trust.trust_still_involved_in_overseas_entity %}
+    {% set isNottrustStillInvolved = trustStillInvolved === "No" %}
```
Comment (markpit) [inline views/includes/check-your-answers/trusts.html:22]: isNottrustStillInvolved needs a capital T on trust

File: test/controllers/update/check.your.answers.controller.spec.ts (hunk 1 above)
Comment (mattch1) [reply test/controllers/update/check.your.answers.controller.spec.ts:1051]: Updated title.

File: views/includes/check-your-answers/trusts.html (hunk 2 above)
Comment (mattch1) [reply views/includes/check-your-answers/trusts.html]: Changed the name slightly while adding this change.

File: views/includes/check-your-answers/individual-trustee.html [hunk 3]
```diff
@@ -209,7 +209,7 @@
   }), rows) %}
 {% endif %}
 
-{% if not pageParams.isRegistration %}
+{% if not pageParams.isRegistration and (trustStillInvolved === "Yes" or isNotStillInvolved) %}
```
Comment (mwestacott) [inline views/includes/check-your-answers/individual-trustee.html:212]: We set this constant up on line 48 in this file:

`{% set isNotStillInvolved = stillInvolved === "No" %}`

Maybe we should set a similar one up for 'still involved'?

I'm also wondering about the check - we see if it's 'yes' or 'no' but I assume that's excluding the case of it being null/undefined?

Comment (mattch1) [reply views/includes/check-your-answers/individual-trustee.html]: Updated this to be a variable. With these we are testing for a specific yes/no value.

Comment (mattch1) [reply views/includes/check-your-answers/individual-trustee.html]: Correct.

**Review by mwestacott — APPROVED** [review]

**Review by dlloyd1-ch — APPROVED** [review]

**Pull request merged**